src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -266,7 +280,18 @@ async Task<DurableHttpResponse> IDurableOrchestrationContext.CallHttpAsync(Durab
                 }
 
                 this.IncrementActionsOrThrowException();
-                await this.InnerContext.CreateTimer(fireAt, CancellationToken.None);
+
+                if (req.Timeout == null)
+                {
+                    await this.InnerContext.CreateTimer(fireAt, CancellationToken.None);
+                }
+                else
+                {
+                    TimeSpan timeLeft = req.TimeoutExpirationDateTime - this.InnerContext.CurrentUtcDateTime;
+                    CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+                    cancellationTokenSource.CancelAfter(timeLeft);
+                    await this.InnerContext.CreateTimer(fireAt, cancellationTokenSource.Token);
```

Comment:
Rather than relying on a cancellation token, would it make sense to instead use the minimum of either `fireAt` or `req.TimeoutExpirationDateTime` as the parameter to `CreateTimer`?